Source/JavaScriptCore/runtime/ArrayPrototype.cpp

```diff
@@ -1523,4 +1523,59 @@ JSC_DEFINE_HOST_FUNCTION(arrayProtoPrivateFuncAppendMemcpy, (JSGlobalObject* glo
     return JSValue::encode(jsUndefined());
 }
 
+JSC_DEFINE_HOST_FUNCTION(arrayProtoPrivateFuncFromFast, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSValue constructor = callFrame->uncheckedArgument(0);
+    if (UNLIKELY(constructor != globalObject->arrayConstructor()))
```

Comment:
Drive-by from someone who is not a JSC contributor, feel free to ignore.

[`Array.from`](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-array.from) is unusual among constructor-respecting static methods in that it can also be called without a `this`, like `(0, Array.from)([0, 1, 2])`. This comes up occasionally in cases like `Promise.resolve([0, 1]).then(Array.from)`.

If I'm reading this right, these patterns will not hit this fast path. It could if this check was something like

```suggestion
    if (UNLIKELY(constructor != globalObject->arrayConstructor() && constructor != undefined))
```

(not that literal code, I don't know how you check for the `undefined` value here, but hopefully you get the idea)

Reply:
This is nice. Changed.